### PR TITLE
Add chat client options

### PR DESCRIPTION
**Pull request by Efnilite**
### Description
- Adds chat visibility expression and type
- Adds condition for whether the player has chat filtering enabled
- Adds condition for whether the player has chat colours enabled

---
**Target Minecraft Versions:** any 
**Requirements:** Paper 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/conditions/CondChatColors.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	@Override
+	public boolean check(Player player) {
+		return player.getClientOption(ClientOption.CHAT_COLORS_ENABLED);
+	}
+
+	@Override
+	protected String getPropertyName() {
+		return "can see chat colors";
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/conditions/CondChatColors.java:36]:
```suggestion
		return "see chat colors";
```

File: src/main/java/ch/njol/skript/conditions/CondChatFiltering.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	@Override
+	public boolean check(Player player) {
+		return player.getClientOption(ClientOption.TEXT_FILTERING_ENABLED);
+	}
+
+	@Override
+	protected String getPropertyName() {
+		return "has chat filtering enabled";
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/conditions/CondChatFiltering.java:34]:
```suggestion
		return "chat filtering enabled";
```

File: src/main/resources/lang/default.lang [hunk 3]
```diff
@@ -2275,6 +2275,13 @@ transform reasons:
 	unknown: unknown
 	infection: infection, villager infection
 
+# -- Chat Visibilities --
+chat visibilities:
+	full: full, visible
```
Comment (sovdeeth) [inline src/main/resources/lang/default.lang:2280]:
```suggestion
	full: fully visible, visible
```

File: src/main/resources/lang/default.lang [hunk 4]
```diff
@@ -2275,6 +2275,13 @@ transform reasons:
...
+	hidden: hidden
+	system: system, commands only, only commands
```
Comment (sovdeeth) [inline src/main/resources/lang/default.lang:2282]:
```suggestion
	system: system messages only, only system messages, commands only, only commands
```

File: src/test/skript/junit/ExprChatVisibility.sk [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+	junit test is {_test}
+
+	if player's chat visibility is only commands:
+		complete objective "chat visibility is only commands" for {_test}
+
+	if player doesn't have chat filtering enabled:
+		complete objective "no chat filtering" for {_test}
+
+	if player can see chat colours:
+		complete objective "can see chat colours" for {_test}
```
Comment (sovdeeth) [inline src/test/skript/junit/ExprChatVisibility.sk:18]:
```suggestion
		complete objective "can see chat colours" for {_test}

```

File: src/test/skript/junit/ExprChatVisibility.sk [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+	set {_tests::1} to "chat visibility is only commands"
+	set {_tests::2} to "no chat filtering"
+	set {_tests::3} to "can see chat colours"
+	ensure junit test "org.skriptlang.skript.test.tests.syntaxes.expressions.ExprChatVisibilityTest" completes {_tests::*}
+
+on join:
+	set {_test} to "org.skriptlang.skript.test.tests.syntaxes.expressions.ExprChatVisibilityTest"
+	junit test is {_test}
+
+	if player's chat visibility is only commands:
```
Comment (sovdeeth) [inline src/test/skript/junit/ExprChatVisibility.sk:11]: honestly seeing this syntax makes me think this should be a custom condition. I'd rather see `if player can only see commands in chat` or something to that effect

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/classes/data/BukkitClasses.java [hunk 7]
```diff
@@ -30,6 +30,7 @@
 import java.util.stream.Collectors;
 
 import ch.njol.skript.bukkitutil.BukkitUtils;
+import com.destroystokyo.paper.ClientOption;
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/classes/data/BukkitClasses.java:33]: this file doesn't need to be edited

File: src/main/java/ch/njol/skript/conditions/CondChatVisibility.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+	public String toString(@Nullable Event event, boolean debug) {
+		return switch (pattern) {
+			case 0 -> player.toString(event, debug) + " can see all messages";
+			case 1 -> player.toString(event, debug) + " can only see commands";
+			case 2 -> player.toString(event, debug) + " can't see any messages";
+			case 3 -> player.toString(event, debug) + " can't see all messages";
+			case 4 -> player.toString(event, debug) + " can't only see commands";
+			default -> throw new IllegalStateException("Unexpected value: " + pattern);
+		};
+	}
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/conditions/CondChatVisibility.java:83]:
```suggestion
	}

```

**Review by sovdeeth — APPROVED** [review]

File: src/main/java/ch/njol/skript/conditions/CondChatVisibility.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+@Name("Can See Messages")
+@Description("Checks whether a player can see specific message types in chat.")
+@Examples({
+	"if player can see all messages:",
+		"\tsend \"You can see all messages.\"",
+	"if player can only see commands:",
+		"\tsend \"This game doesn't work with commands-only chat.\"",
+	"if player can't see any messages:",
+		"\tsend action bar \"Server shutting down in 5 minutes!\""
+})
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/conditions/CondChatVisibility.java:26]:
```suggestion
})
@RequiredPlugins("Paper")
```

**Review by Moderocky — APPROVED** [review]

**Pull request merged**